python/ray/llm/_internal/serve/deployments/utils/node_initialization_utils.py

```diff
@@ -93,29 +93,34 @@ async def initialize_node(llm_config: LLMConfig) -> InitializeNodeOutput:
     pg = engine_config.get_or_create_pg()
     runtime_env = engine_config.get_runtime_env_with_local_env_vars()
 
-    if engine_config.placement_strategy == "STRICT_PACK":
-        # If the placement strategy is STRICT_PACK, we know that all the
-        # workers run on the same node as the engine. Therefore, we can run
-        # all initialization steps directly instead of in tasks in the PG.
-        # This removes the task launching overhead reducing the initialization
-        # time.
-        local_node_download_model = local_node_download_model.union(
-            worker_node_download_model
-        )
+    # If the load format is runai_streamer, we don't need to do any model download.
+    if llm_config.engine_kwargs.get("load_format", "") not in (
+        "runai_streamer",
+        "runai_streamer_sharded",
+    ):
```

Comment:
I think this is very hacky. We want to unify how this is handled and how it will be handled on ray data. See this PR: https://github.com/ray-project/ray/pull/56906/files#diff-67e711e81059eab3f10cf494080e597955ced03c9dcd4dc7ef18a304d0f58df0


This PR uses the download_model model enum to control what to download and what to skip, we should expose that through download_files util and use that. It might require a little bit of refactoring to make it nice.